hazelcast-client-legacy/src/main/java/com/hazelcast/client/impl/ClusterAuthenticator.java

```diff
@@ -60,7 +60,7 @@ public void authenticate(ClientConnection connection) throws AuthenticationExcep
         //contains remoteAddress and principal
         SerializableList collectionWrapper;
         final ClientInvocation clientInvocation = new ClientInvocation(client, auth, connection);
-        final Future<SerializableList> future = clientInvocation.invoke();
+        final Future<SerializableList> future = clientInvocation.invokeUrgent();
```

Comment:
I know, this is why I have made this comment under `hazelcast-client-legacy` not `hazelcast-client`